### DISCUSSION TITLE
Surefire to CTRF report converter to convert the surefire reports to CTRF Format

### DISCUSSION THREAD

**Discussion by sahilwingify** [ctrf-io/ctrf#18] [Creations]
@Ma11hewThomas I have created a testng to CTRF converter by creating a fork of Junit to ctrf converter. Should we publish that as well?

**Comment by sahilwingify** [comment]
https://github.com/sahilwingify/testng-to-ctrf

here is the forked repo that converts testng-results.xml file to CTRF format

**Reply by sahilwingify** [reply]
If this looks good to you, we can publish this repo as well

**Reply by sahilwingify** [reply]
@Ma11hewThomas Could you please check this?

**Comment by Ma11hewThomas** [comment]
Thank you for building this plugin @sahilwingify! We’d be thrilled if you decided to publish it. Adding a section for community-built plugins is on our roadmap for the website and this plugin would be featured. 

I don’t have experience with Surefire myself, but the code looks well-structured overall. You may want to consider the plugin’s name to surefire-to-ctrf, as, if I understand correctly, Surefire reports can be generated by various Java-based frameworks, including JUnit — though please correct me if I'm mistaken.

**Reply by sahilwingify** [reply]
@Ma11hewThomas, yeah, surefire-to-ctrf does make sense. I'll make the changes and let you know once I'm done. If I may ask, what is the plan to publish it?

**Reply by sahilwingify** [reply]
@Ma11hewThomas I have updated the name of the repo and also updated the code to handle errors better

https://github.com/sahilwingify/surefire-to-ctrf

Please have a look and suggest when we can publish this.

**Comment by Ma11hewThomas** [comment]
Thank you for creating this @sahilwingify! You can publish the package to npm yourself, and we’ll include it on our website as a community-built plugin once the plugin page is live. Alternatively, you can transfer full ownership of the code to the CTRF io organization, and we’ll handle the publishing. In that case, we’ll add you as a repository contributor.

If you opt for the latter, we will need to discuss the transfer process further.

Thanks again!

**Comment by sahilwingify** [comment]
@Ma11hewThomas Thanks, will publish it on NPM and let you know. Just to add the file conversion will work for Testng reports only and not for surefire reports.

So will name it as testng-to-ctrf convertor

**Comment by R-Nordberg** [comment]
@sahilwingify  What happened the to the repo / uploading to NPM? I can't seem to find testng-to-ctrf anywhere on GitHub or NPM

**Comment by Ma11hewThomas** [comment]
@R-Nordberg the [junit-to-ctrf](https://github.com/ctrf-io/junit-to-ctrf) now supports conversion of testng surefire. Let me know if this works for you.
